### PR TITLE
Update for modified settings in ST4

### PR DESCRIPTION
**Pull request by rwols**

### CODE REVIEW

**Comment by rchl** [comment]
I have a feeling there might be more changes. Have you based this just on the ones that show errors?

**Comment by rwols** [comment]
> Have you based this just on the ones that show errors?

Yes.

**Comment by rwols** [comment]
I went over the default settings. I think this is everything :)

**Comment by rwols** [comment]
I missed light_color_scheme and dark_color_scheme

**Comment by rchl** [comment]
Missing:
- `auto_complete_when_likely`
- `mini_auto_complete`
- `remember_workspace`
(not all are documented unfortunately)

Change:
 - `draw_unicode_white_space` should be have `enum`s defined
 - `auto_find_in_selection` supports more options

File: schemas/preferences.sublime-settings.json [hunk 1]
```diff
@@ -11,7 +11,19 @@
         "color_scheme": {
           "markdownDescription": "Sets the colors used within the text area.",
           "type": "string",
-          "pattern": "^(?:(.+\\.((sublime|hidden)-color-scheme|tmTheme)|auto)$",
+          "pattern": "^(.+\\.((sublime|hidden)-color-scheme|tmTheme)|auto)",
+          "default": "Mariana.sublime-color-scheme"
```
Comment (rchl) [inline schemas/preferences.sublime-settings.json:15]: The default is "auto" I believe.

File: schemas/preferences.sublime-settings.json [hunk 2]
```diff
@@ -824,7 +866,7 @@
         "theme": {
           "markdownDescription": "The theme controls the look of Sublime Text's UI (buttons, tabs, scroll bars, etc).",
           "type": "string",
-          "pattern": "^(?:.*\\.sublime-theme|auto)$",
+          "pattern": "^(.*\\.sublime-theme|auto)$",
           "default": "Default.sublime-theme"
```
Comment (rchl) [inline schemas/preferences.sublime-settings.json:870]: "auto"

File: schemas/preferences.sublime-settings.json (hunk 1 above)
Comment (rwols) [reply schemas/preferences.sublime-settings.json:15]: The default is Mariana.sublime-color-scheme.

File: schemas/preferences.sublime-settings.json (hunk 2 above)
Comment (rwols) [reply schemas/preferences.sublime-settings.json]: fixed in 3d193ef

File: schemas/preferences.sublime-settings.json [hunk 3]
```diff
@@ -11,7 +11,19 @@
         "color_scheme": {
           "markdownDescription": "Sets the colors used within the text area.",
           "type": "string",
-          "pattern": "^.+\\.((sublime|hidden)-color-scheme|tmTheme)$",
+          "pattern": "^(.+\\.((sublime|hidden)-color-scheme|tmTheme)|auto)",
```
Comment (rchl) [inline schemas/preferences.sublime-settings.json:14]: Is `$` at the end gone on purpose?

File: schemas/preferences.sublime-settings.json [hunk 4]
```diff
@@ -11,7 +11,19 @@
...
+          "default": "Mariana.sublime-color-scheme"
+        },
+        "light_color_scheme": {
+          "markdownDescription": "Controls the `\"color_scheme\"` when set to `\"auto\"`.",
+          "type": "string",
+          "pattern": "^.+\\.((sublime|hidden)-color-scheme|tmTheme)",
+          "default": "Breakers.sublime-color-scheme"
+        },
+        "dark_color_scheme": {
+          "markdownDescription": "Controls the `\"color_scheme\"` when set to `\"auto\"`.",
```
Comment (rchl) [inline schemas/preferences.sublime-settings.json:24]: I know this is copied from default preferences but I think it should say "... when `color_scheme` is set to "auto""

File: schemas/preferences.sublime-settings.json (hunk 3 above)
Comment (rwols) [reply schemas/preferences.sublime-settings.json:14]: I was under the impression that the $ didn't matter, but it does after double checking

Schermafbeelding 2021-05-23 om 23 35 47

vscode-json-language-server should display a warning here. Let me fix that.

**Pull request merged**